cmd/gangue/gangue.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+
+	"github.com/coreos/mantle/auth"
+	"github.com/coreos/mantle/cli"
+	"github.com/coreos/mantle/sdk"
+)
+
+var (
+	root = &cobra.Command{
+		Use:   "gangue",
+		Short: "Google Storage download and verification tool",
+	}
+
+	gpgKeyFile, jsonKeyFile string
+	serviceAuth, verify     bool
+)
+
+func validateGSURL(rawURL string) error {
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return err
+	}
+	if parsedURL.Scheme != "gs" {
+		return fmt.Errorf("URL missing gs:// scheme: %v", rawURL)
+	}
+	if parsedURL.Host == "" {
+		return fmt.Errorf("URL missing bucket name %v", rawURL)
+	}
+	if parsedURL.Path == "" {
+		return fmt.Errorf("URL missing file path %v", rawURL)
+	}
+	if parsedURL.Path[len(parsedURL.Path)-1] == '/' {
+		return fmt.Errorf("URL must not be a directory path %v", rawURL)
+	}
+	return nil
+}
+
+func run(cmd *cobra.Command, args []string) {
+	var client *http.Client
+	output, source := "", ""
+
+	if len(args) == 2 {
+		source = args[0]
+		output = args[1]
+	} else if len(args) == 1 {
+		source = args[0]
+		output = "."
+	} else {
+		fmt.Fprintf(os.Stderr, "Expected one or two arguments\n")
+		os.Exit(1)
+	}
+
+	// Perform some basic sanity checks on the options
+	if output == "" {
+		fmt.Fprintf(os.Stderr, "The output file path is required\n")
```

Comment:
This seems confusing when it is actually optional, perhaps just convert a blank arg to "."